packages/core/src/renderer/components/+helm-releases/release-details/release-details-model/release-details-model.injectable.tsx

```diff
@@ -50,9 +54,22 @@ const releaseDetailsModelInjectable = getInjectable({
       toHelmRelease: di.inject(toHelmReleaseInjectable),
     });
 
-    await model.load();
-
-    return model;
+    return asyncComputed<ReleaseDetailsModel>({
+      getValueFromObservedPromise: async () => {
+        const currentHelmRelease = detailsHelmRelease.get();
+
+        if (
+          currentHelmRelease
+          && currentHelmRelease.name === targetRelease.name
+          && currentHelmRelease.namespace === targetRelease.namespace
+        ) {
+          await model.load();
+        }
+
+        return model;
+      },
+      betweenUpdates: "show-latest-value",
+    });
   },
 
   lifecycle: lifecycleEnum.keyedSingleton({
```

Comment:
Would we lose anything substantial if `lifecycle` here was `lifecycleEnum.transient`, and `instantiate` was `async` because of `await model.load()`? Then we wouldn't need `asyncComputed` at all, and `withInjectables` could be responsible of rendering a spinner while an async dependency (ie. a fresh instance of `ReleaseDetailsModel`) is being instantiated.

Reply:
Proposed solution would be simpler with less code, and less prone to bugging. Eg. existing solution in PR will likely still show stale data when reopening same "release details" after a long wait, as that will not cause new `model.load()`. And it will briefly show details of wrong release when changing to other release, because of `betweenUpdates: "show-latest-value"`.